modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/index.d.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+/* Debounces function execution. */
+export function debounce(fn: () => void, delay: number): () => void;
+
+/* Cancels the scheduled debounced function. */
+export function cancelDebounce(debounced: () => void): void;
+
+/**
+ * Listens to the specified event on the given DOM element, but only calls the
+ * given callback listener when it's triggered by elements that match the
+ * given selector or target element.
+ */
```

Comment:
Are these comments for the TS hints?  I don't think it is super necessary to add these comments, unless we want to directly copy from the comment where the source file is.

Reply:
Essentially, yes. I moved them from the JS file to the TS definition file. As I went with the assumption that JSDoc is no longer necessary when we had the type definitions. But I like the little snippet explaining what the function does, so I kept that part in. What're your thoughts on that?

Reply:
I think it is fine to copy them over from the source(still keeping them in the source too), but I wouldn't spend too much time trying to write descriptions for everything.

Reply:
I agree, I've been adding the ones that already exist, just copying them over, and made a couple new ones that took me under a minute to think of

Reply:
> Are these comments for the TS hints? I don't think it is super necessary to add these comments, unless we want to directly copy from the comment where the source file is.

I also agree, I was thinking that with TypeScript we wouldn't really need these comments anyway